object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -338,44 +339,135 @@ private void handleAssignment(AssignmentNode node, Set<LocalVarWithTree> newDefs
     if (lhsElement.getKind().equals(ElementKind.FIELD) && typeFactory.hasMustCall(lhs.getTree())) {
       if (rhs instanceof LocalVariableNode && isVarInDefs(newDefs, (LocalVariableNode) rhs)) {
         if (typeFactory.getDeclAnnotation(lhsElement, Owning.class) != null) {
-          LocalVarWithTree latestAssignmentPair =
-              getAssignmentTreeOfVar(newDefs, (LocalVariableNode) rhs);
-          newDefs.remove(latestAssignmentPair);
+          Set<LocalVarWithTree> setContainsLatestAssignmentPair =
+              getSetContainsAssignmentTreeOfVar(newDefs, (LocalVariableNode) rhs);
+          newDefs.remove(setContainsLatestAssignmentPair);
         }
       }
     } else if (lhs instanceof LocalVariableNode
         && !isTryWithResourcesVariable((LocalVariableNode) lhs)) {
 
       // Reassignment to the lhs
       if (isVarInDefs(newDefs, (LocalVariableNode) lhs)) {
+        Set<LocalVarWithTree> setContainsLatestAssignmentPair =
+            getSetContainsAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
         LocalVarWithTree latestAssignmentPair =
             getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
-        checkMustCall(
-            latestAssignmentPair,
-            typeFactory.getStoreBefore(node),
-            "variable overwritten by assignment " + node.getTree());
-        newDefs.remove(latestAssignmentPair);
+        if (setContainsLatestAssignmentPair.size() == 1) {
+          checkMustCall(
+              newDefs,
+              setContainsLatestAssignmentPair,
+              typeFactory.getStoreBefore(node),
+              "variable overwritten by assignment " + node.getTree());
+          newDefs.remove(setContainsLatestAssignmentPair);
+        } else {
+          setContainsLatestAssignmentPair.remove(latestAssignmentPair);
```

Comment:
This mutation of `setContainsLatestAssignmentPair` is dangerous!  It's going to change the set everywhere, including any place it is stored in the `visited` set.  I think we need to change `BlockWithLocals` to only store `ImmutableSet`s, and then create a fresh set here with the value removed